website/static/js/comment.js

```diff
@@ -242,34 +371,109 @@ var CommentModel = function(data, $parent, $root) {
         self.unreporting, self.undeleting
     );
 
-    self.isVisible = ko.computed(function() {
-        return !self.isDeleted() && !self.isAbuse();
+    self.isVisible = ko.pureComputed(function() {
+        return !self.isDeleted() && !self.isHidden() && !self.isAbuse();
     });
 
-    self.editNotEmpty = ko.computed(function() {
+    self.editNotEmpty = ko.pureComputed(function() {
         return notEmpty(self.content());
     });
 
-    self.toggleIcon = ko.computed(function() {
-            return self.showChildren() ? 'fa fa-minus-square-o' : 'fa fa-plus-square-o';
+    self.toggleIcon = ko.pureComputed(function() {
+        return self.showChildren() ? 'fa fa-minus-square-o' : 'fa fa-plus-square-o';
     });
-    self.editHighlight = ko.computed(function() {
-        return self.canEdit() && self.hoverContent();
+    self.editHighlight = ko.pureComputed(function() {
+        return self.canEdit() && self.hoverContent() && self.mode !== 'widget';
     });
-    self.canReport = ko.computed(function() {
+    self.canReport = ko.pureComputed(function() {
         return self.$root.canComment() && !self.canEdit();
     });
 
-    self.shouldShow = ko.computed(function() {
-        return !self.isDeleted() || self.hasChildren() || self.canEdit();
+    self.shouldShow = ko.pureComputed(function() {
+        if (!self.isDeleted() && !self.isHidden()) {
+            return true;
+        }
+        if (self.isHidden()) {
+            return self.level === 1;
+        }
+        return self.hasChildren() || self.canEdit();
+    });
+
+    self.shouldShowChildren = ko.pureComputed(function() {
```

Comment:
This cannot be a `pureComputed` (it has the side effect of setting `self.showChildren`.